### PR TITLE
Add Zcoin Support

### PR DESCRIPTION
**Pull request by josephnicholas**
### Add Zcoin support for BATM

### CODE REVIEW

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public IRateSource createRateSource(String sourceLogin) {
+        if (sourceLogin != null && !sourceLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(sourceLogin, ":");
+            String exchangeType = st.nextToken();
+            if ("coinmarketcap".equalsIgnoreCase(exchangeType)) {
+                String preferredFiatCurrency = Currencies.USD;
+                if (st.hasMoreTokens()) {
+                    preferredFiatCurrency = st.nextToken().toUpperCase();
+                }
+                return new CoinexchangeRateSource(preferredFiatCurrency);
```
Comment (rpanak-generalbytes) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java]: you are using `"coinmarketcap"` prefix for `CoinexchangeRateSource`?

**Comment by rpanak-generalbytes** [comment]
@josephnicholas To be clear about the review - exchange support is not required for the merge to happen. It's just a suggestion.

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java (hunk 1 above)
Comment (josephnicholas) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java]: Yes, this means I am using coinmarketcap as exchange for price right? I just followed the procedure on other coins.

Comment (rpanak-generalbytes) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java]: No. You are instantiating `CoinexchangeRateSource`. No coinmarketcap usage in your code as far as I can see.

Comment (josephnicholas) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java]: Okay is there a better way to implement this? Apologies, I thought there is already an API handling this

Comment (rpanak-generalbytes) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/zcoin/ZcoinExtension.java]: No problem. Just instantiate a `CoinmarketcapRateSource` instead (see for example `ElectraExtension`).

**Comment by josephnicholas** [comment]
> @josephnicholas To be clear about the review - exchange support is not required for the merge to happen. It's just a suggestion.

Code updated @rpanak-generalbytes . Please review.

**Comment by rpanak-generalbytes** [comment]
@josephnicholas You can make your choice based on this photo of the home screen. It is very close to the default BATM UI. The theme is customizable, so there might be different colors. In my opinion the two circular ones work the best. 
cf0c5d99-bfd7-4dea-bd67-d9bee2f06161

**Comment by josephnicholas** [comment]
We have a SVG file of the image below, will include this in our PR if approved by you @rpanak-generalbytes .

zcoin-circular-background

@joernroeder

**Comment by josephnicholas** [comment]
Done @rpanak-generalbytes

**Comment by rpanak-generalbytes** [comment]
@josephnicholas Thanks for the PR.

**Pull request merged**